apps/prairielearn/src/lib/question-render.ts

```diff
@@ -606,6 +607,8 @@ export async function renderPanelsForSubmission({
   questionContext: QuestionContext;
   authorizedEdit: boolean;
   renderScorePanels: boolean;
+  // groupRolePermissions is only used for student instance question pages
+  groupRolePermissions?: { can_view: boolean; can_submit: boolean } | null;
```

Comment:
At least with `authorizedEdit` and `renderScorePanels`, I made the deliberate decision to make them required (that is, they can't be `undefined`) so that we're forced to always be explicit with what we provide. I'd be happy to make that change here too, though certainly won't block on it. I'll leave that choice to you.

Reply:
Done in 7ccff4a518e6a2ea6ffbed27a93756be1bc34dbc.